stream-chat-android-sample/src/main/kotlin/io/getstream/chat/sample/application/ChatInitializer.kt

```diff
@@ -21,8 +29,17 @@ class ChatInitializer(private val context: Context) {
             )
         val notificationHandler = SampleNotificationHandler(context, notificationConfig)
 
-        val client = ChatClient.Builder(apiKey, context).loggerHandler(FirebaseLogger).notifications(notificationHandler).logLevel(ChatLogLevel.ALL).build()
-        val domain = ChatDomain.Builder(client, context).offlineEnabled().notificationConfig(notificationConfig).build()
-        val ux = ChatUI.Builder(client, domain, context).build()
+        client =
+            ChatClient.Builder(apiKey, context).loggerHandler(FirebaseLogger).notifications(notificationHandler)
+                .logLevel(ChatLogLevel.ALL).build()
+
+        domain =
+            ChatDomain.Builder(client!!, context).offlineEnabled().notificationConfig(notificationConfig).build()
+        ux = ChatUI.Builder(client!!, domain!!, context).build()
+    }
+
+    private fun disconnectChatsIfNecessary() {
+        client?.disconnect()
+        GlobalScope.launch(Dispatchers.Main) { domain?.disconnect() }
```

Comment:
i don't think this last disconnect is needed, the domain follows the .disconnect() call from the client. https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt#L280

Reply:
Code can be simplified using: https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt#L945